src/librustc_mir/transform/deaggregator.rs

```diff
@@ -130,7 +130,7 @@ fn get_aggregate_statement_index<'a, 'tcx, 'b>(start: usize,
         debug!("getting variant {:?}", variant);
         debug!("for adt_def {:?}", adt_def);
         let variant_def = &adt_def.variants[variant];
-        if variant_def.kind == VariantKind::Struct {
+        if variant_def.ctor_kind == CtorKind::Fictive {
```

Comment:
Yeah this looks like it can only limit the optimization, it should probably be removed. cc @pcwalton